extensions/ql-vscode/src/debugger/debug-session.ts

```diff
@@ -155,7 +260,9 @@ export class QLDebugSession extends LoggingDebugSession implements Disposable {
   }
 
   public dispose(): void {
-    this.cancelEvaluation();
+    if (this.runningQuery !== undefined) {
+      this.runningQuery.cancel();
+    }
```

Comment:
Oh....I see you did here exactly what I suggested above. Can you change this so `this.runningQuery` is disposed here and in the `RunningQuery` class itself, the dispose method calls cancel?

Reply:
I'm not sure it's OK to dispose the token source until we're sure the cancellation has actually been observed by the running query. The way I've implemented it, the running query (and its token source) are disposed after query completes (even if cancelled), and the debug session just "lets go" of the running query.